ballerina/bytes_to_sse_stream_generator.bal

```diff
@@ -0,0 +1,168 @@
+                if nextByte is () {
+                    return ();
+                }
+                if nextByte == LINE_FEED {
+                    buffer.push(currentByte);
+                    buffer.push(nextByte);
+                    return string:fromBytes(buffer);
+                }
+                // Store char in lookahead buffer if not newline
+                self.lookaheadBuffer.push(nextByte);
+            }
+            foundCariageReturnWithNewLine = false;
+            if (currentByte == LINE_FEED || currentByte == CARRIAGE_RETURN) && prevByte == currentByte {
+                buffer.push(currentByte);
+                return string:fromBytes(buffer);
+            }
+            if currentByte == LINE_FEED && prevByte == CARRIAGE_RETURN {
+                foundCariageReturnWithNewLine = true;
+            }
+            buffer.push(currentByte);
+            prevByte = currentByte;
+        }
+        return ();
+    }
+
+    # Reads next byte from the lookahead buffer if data is available, otherwise read from the byte stream
+    # + return - A `byte?` on success, `error` on failure
+    private isolated function getNextByte() returns byte|error? {
+        if self.lookaheadBuffer.length() > 0 {
+            return self.lookaheadBuffer.shift();
+        }
+        record {byte[] value;}? nextValue = check self.byteStream.next();
+        return nextValue is () ? () : nextValue.value[0];
+    }
+}
+
+isolated function parseSseEvent(string event) returns SseEvent|error {
+    string[] lines = re `\r\n|\n|\r`.split(event);
+    string? id = ();
+    string? comment = ();
+    string? data = ();
+    int? 'retry = ();
+    string? eventName = ();
+
+    foreach string line in lines {
+        if line == "" {
+            continue;
+        }
+        regexp:Groups? groups = re `(.*?):(.*)`.findGroups(line);
+        string filedName = line;
+        string fieldValue = "";
+        if groups is regexp:Groups && groups.length() == 3 {
+            regexp:Span? filedNameSpan = groups[1];
+            regexp:Span? filedValueSpan = groups[2];
+            if filedNameSpan is () || filedValueSpan is () {
+                continue;
+            }
+            filedName = filedNameSpan.substring().trim();
+            fieldValue = removeLeadingSpace(filedValueSpan.substring());
+        }
+        if filedName == ID {
+            id = fieldValue;
+        } else if filedName == COMMENT {
+            comment = fieldValue;
+        } else if filedName == RETRY {
+            int|error retryValue = int:fromString(fieldValue);
+            'retry = retryValue is error ? () : retryValue;
+        } else if filedName == EVENT {
+            eventName = fieldValue;
+        } else if filedName == DATA {
+            if data is () {
+                data = fieldValue;
+            } else {
+                data += fieldValue;
+            }
+        }
+    }
+    SseEvent sseEvent = {data};
```

Comment:
All the fields are optional right, can't we construct the entire event directly?

Reply:
There is no direct way right? Is there a way to do that? 
Anyway  constructing with `{data}` here is wrong we need to have a if check for `data` field as well. Will fix this.

Reply:
Since the fields are only optional(not nilable) we can directly generate the record from the fields

Reply:
Updated to construct them directly.